test/auto_parallel/spmd_rules/test_reshape_rule.py

```diff
@@ -0,0 +1,185 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from paddle.distributed.auto_parallel.static.completion import get_spmd_rule
+from paddle.distributed.auto_parallel.static.dist_attribute import (
+    DistTensorSpec,
+    TensorDistAttr,
+)
+from paddle.distributed.fleet import auto
+
+
+class TestReshapeSPMDRule(unittest.TestCase):
+    def setUp(self):
+        self.rule = get_spmd_rule("reshape")
+
+        x_shape = [6, 12, 48, 24]
+        process_mesh = auto.ProcessMesh(mesh=[[0, 1, 2], [3, 4, 5]])
+
+        x_tensor_dist_attr = TensorDistAttr()
+        x_tensor_dist_attr.dims_mapping = [-1, -1]
+        x_tensor_dist_attr.process_mesh = process_mesh
+        self.x_dist_tensor_spec = DistTensorSpec(x_shape, x_tensor_dist_attr)
+
+        self.attrs = {"shape": [1, 72, 48, 4, 6]}
+
+    def test_reshape_infer_forward(self):
```

Comment:
unitest should include following cases:
1. input axis directly map to output axis 
2. multiple input axes merge into single output axis, with shard on first input axis
3. multiple input axes merge into single output axis, with shard on axis other than first input axis
4. single input axis split into multiple output axes, with first output axis dividable
5. single input axis split into multiple output axes, with first output axis non-dividable
6. multiple input axes transform into  multiple output axis, with shard on first input axis/shard on input axis other than the first axis/ first output axis dividable/ first output axis non-dividable

Reply:
Done